dotty-tensorflow/src/main/scala/me/shadaj/scalapy/tensorflow/api/Variable.scala

```diff
@@ -3,8 +3,12 @@ package me.shadaj.scalapy.tensorflow.api
 import me.shadaj.scalapy.tensorflow.{Variable => PyVariable}
 import scalaUtils.PythonType
 import scala.language.implicitConversions
+import Tensor._
 
-class Variable private[api] (override val underlying: PyVariable) extends Tensor(underlying) {}
+class Variable private[api] (override val underlying: PyVariable) extends Tensor(underlying) {
+  def readValue(): Tensor = underlying.read_value()
```

Comment:
Hmm this might be somewhat confusing... since `Variable` itself extends `Tensor`, what's the difference b/w upcast (`variable.asInstanceOf[Tensor]`) and `variable.readValue()`?

Reply:
I think I get what you talking about. [link](https://medium.com/the-artificial-impostor/notes-understanding-tensorflow-part-1-5f0ebb253ad4#:~:text=Tensors%20v.s.%20Variables&text=It%20represents%20a%20node%20in,graph%20and%20optionally%20its%20gradients.&text=A%20variable%20in%20Tensorflow%20is,and%20changeable%20across%20different%20Session.) 
As far as I understand Variable is a wraper for tensor that proxies most operations, but not `read_value`  because that gets the tensor out.